src/utils.ts

```diff
@@ -1,4 +1,63 @@
-import picomatch from 'picomatch';
+import picomatch, { type Matcher } from 'picomatch';
+
+// #region PARTIAL MATCHER
+export interface PartialMatcherOptions {
+  dot?: boolean;
+  nocase?: boolean;
+}
+
+// the result of over 4 months of figuring stuff out and a LOT of help
+export function getPartialMatcher(patterns: string[], options?: PartialMatcherOptions): Matcher {
+  const regexes = patterns.map(pattern => splitPattern(pattern).map(part => picomatch.makeRe(part, options)));
+  return (input: string) => {
+    // no need to `splitPattern` as this is indeed not a pattern
+    const inputParts = input.split('/');
+    for (let i = 0; i < patterns.length; i++) {
+      const patternParts = splitPattern(patterns[i]);
```

Comment:
Aren't we duplicating some work on lines 11 (`patterns.map(pattern => splitPattern(pattern)...`) and 16?

Reply:
looks like we are, will fix